File: Pset 1/Cash.c
//greedy algorithm

#include <stdio.h>
#include <math.h>
#include <cs50.h>

int main(void)
{
    float received;

    //prompt for change
    do
    {
        received = get_float();
    }
    while(received <0);

    int cents = round(100*received);
    int coins = 0;

    for (int i = 25; i > 0; i--)
    {
        if (i == 25 || i == 10 || i == 5 || i == 1)
        {
            //subtract 25 cents
            while (cents >= i)
            {
                cents -= i;
                coins++;
            }
        }
    }

    //print out
    printf("%i\n", coins);
}

File: Pset 6/cash.py
from cs50 import get_float

received = get_float("Change owed: ")

while True:
    if received > 0:
        break

cents = round(100*received)
coins = 0

array = [25, 10, 5, 1]

for i in range(25, 0, -1):
    if i in array:
        while cents >= i:
            cents -= i
            coins += 1

print(coins)